### PR TITLE
Allow specifying accounts with Address20

### PR DESCRIPTION
**Pull request by nathanwhit**
# Description of proposed changes

TLDR make this work:
Screenshot 2023-12-11 at 2 32 44 PM

We were using the basic `AccountIdLookup` implementation which doesn't do anything special for 20-byte addresses. In our case, though, we can map a 20-byte address using the `AddressMapping` to go from a 20-byte address to our native 32-byte account ID.

---

Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

### CODE REVIEW

**Review by atodorov — APPROVED** [review]

**Review by pLabarta — APPROVED** [review]
Ah, this is great :rocket:
